static/skin/index.js

```diff
@@ -1,62 +1,68 @@
-        url += `${key}=${basicConfig[key]}${idx !== Object.keys(basicConfig).length - 1 ? '&' : ''}`;
-    });
+    url += Object.keys(incrementalLoadingParams).map(key => `${key}=${incrementalLoadingParams[key]}`).join("&");
     return url;
 }
 
+function viewPortToCount(){
+    return Math.floor(window.innerHeight/100 + 1)*(window.innerWidth>1000 ? 3 : 2);
+}
+
 function getInnerHtml(node, query) {
     return node.querySelector(query).innerHTML;
 }
 
-async function loadAndDisplayBooks(append = false) {
+function generateBookHtml(book) {
+    const link = book.querySelector('link').getAttribute('href');
+    const title =  getInnerHtml(book, 'title');
+    const description = getInnerHtml(book, 'summary');
+    const id = getInnerHtml(book, 'id');
+    const iconUrl = getInnerHtml(book, 'icon');
+    const articleCount = getInnerHtml(book, 'articleCount');
+    const mediaCount = getInnerHtml(book, 'mediaCount');
+
+    return `<a href='${link}' data-id='${id}'><div class='book'>
+        <div class='book__background' style="background-image: url('${iconUrl}');">
+        <div class='book__title' title='${title}'>${title}</div>
+        <div class='book__description' title='${description}'>${description}</div>
+        <div class='book__info'>${articleCount} articles, ${mediaCount} medias</div>
+        </div>
+    </div></a>`;
+}
+
+async function loadAndDisplayBooks() {
     isFetching = true;
     fetch(queryUrlBuilder()).then(async (resp) => {
         const data = new window.DOMParser().parseFromString(await resp.text(), 'application/xml');
         const books = data.querySelectorAll('entry');
         let bookHtml = '';
-        books.forEach((book) => {
-            const link = book.querySelector('link').getAttribute('href');
-            const title =  getInnerHtml(book, 'title');
-            const description = getInnerHtml(book, 'summary');
-
-            bookHtml += `<a href='${link}' data-id='${getInnerHtml(book, 'id')}'><div class='book'>
-                <div class='book__background' style="background-image: url('${getInnerHtml(book, 'icon')}');">
-                <div class='book__title' title='${title}'>${title}</div>
-                <div class='book__description' title='${description}'>${description}</div>
-                <div class='book__info'>${getInnerHtml(book, 'articleCount')} articles, ${getInnerHtml(book, 'mediaCount')} medias</div>
-                </div>
-            </div></a>`;
-        });
-        document.querySelector('.book__list').innerHTML = (append ? document.querySelector('.book__list').innerHTML : '') + bookHtml;
-        isEnd = !books.length;
+        books.forEach((book) => {bookHtml += generateBookHtml(book)});
+        document.querySelector('.book__list').innerHTML += bookHtml;
+        incrementalLoadingParams['start'] = incrementalLoadingParams['start'] + incrementalLoadingParams['count'];
+        if (!books.length) {
+            incrementalLoadingParams['count'] = 0;
+        }
```

Comment:
You can set `count` to 0 sooner, if you check that `books.length < incrementalLoadingParams['count']`